server/market/bookrouter.go

```diff
@@ -296,6 +313,11 @@ func (r *BookRouter) Run(ctx context.Context) {
 	wg.Wait()
 }
 
+// LastFeeRate is the last fee rate registered for an asset.
+func (r *BookRouter) LastFeeRate(assetID uint32) uint64 {
+	return atomic.LoadUint64(r.feeRateCache[assetID])
+}
```

Comment:
I had some thoughts about the roles of `BookRouter`, `Market`, and `OrderRouter` with respect to the fee rate fetching, caching, and retrieval that I shared in matrix.  The following is my high level understanding of how the pieces currently fit and what is responsible for what:

The assets implement `market.FeeFetcher`, which is a `FeeRate() (uint64, error)` method. These are provided to `Market`, who uses the fetcher at match time to get fee rates per asset for all the match sets.
These base/quote fee rates make their way the outgoing `BookRouter` via an `epoch_report` signal. The `BookRouter` stores these rates when it gets them.
`BookRouter` also implements `LastFeeRate(assetID uint32) uint64` (this commented line), and `OrderRouter` uses `BookRouter` as a `FeeSource interface` for this method. The `OrderRouter` uses this interface to ensure zero conf funded orders have high enough fee rate on funding txn.
So `BookRouter` with it's fee rate cache is the fee source for both itself and `OrderRouter`, while `Market` uses the assets directly for fee rates.

One question I had was what fee rates are known/cached before the first epoch_report? Also, how can we populate the book router's fee rate cache earlier, ideally construction time?

A general suggestion I had was to reduce and clarify the roles of these three `server/market` subsystems with respect to fee rate fetching/caching.  Presently the flow of fee rates is from RPC fetch in `Market` on epoch close -> fee rate cache in `BookRouter` via epoch_report -> retrieve from cache in `OrderRouter` via `BookRouter`'s `LastFeeRate` method.  This design makes `BookRouter` implement the cache, while `Market` actually does retrieval.  So I'm contemplating if a `FeeSource` can be some other type implemented outside of the market package, like something in server/dex that can do the RPC fetch and cache management without troubling the `server/market` types. Also, `server/dex.DEX` has the backends so it can hit `FeeRate()` to prepopulate the fee rate caches before even firing up the market subsystems.

Part of the appeal of a type that handles fee fetching and caching is consolidating these responsibilities that are now shared between the three market subsystems, and a fetch can automatically update the cache directly.  Both `OrderRouter` and `BookRouter` can get the same `FeeSource`, and calls from `Market` to this type's `FetchFeeRate(assetID uint32)` method (indirectly via wrapper for the relevant assetID) could update the cache.